src/modm/driver/temperature/max31865.hpp

```diff
@@ -0,0 +1,216 @@
+		FaultDetectionCycle0 = Bit2,
+		FaultStatusClear = Bit1,
+		Rejection = Bit0
+	};
+	MODM_FLAGS8(Config)
+
+	enum class FaultDetectionWrite : uint8_t
+	{
+		NoAction = 0b00,
+		FaultDetectionAutomaticDelay = 0b01,
+		RunFaultDetectionWithManualDelayCycle1 = 0b10,
+		FinishFaultDetectionWithManualDelayCycle2 = 0b11,
+	};
+	typedef Configuration<Config_t, FaultDetectionWrite, 0b11, 2>
+		FaultDetectionWrite_t;  // Bit 8..10
+
+	enum class FaultDetectionRead : uint8_t
+	{
+		FaultDetectionFinished = 0b00,
+		AutomaticDetectionStillRunning = 0b01,
+		ManualCycle1Running =
+			0b10,  // waiting for user to write FinishFaultDetectionWithManualDelayCycle2
+		ManualCycle2StillRunning = 0b11,
+	};
+	typedef Configuration<Config_t, FaultDetectionRead, 0b11, 2> FaultDetectionRead_t;  // Bit 8..10
+
+	enum class Rejection : uint8_t
+	{
+		Rejection60Hz = 0b0,
+		Rejection50Hz = 0b1,
+	};
+	typedef Configuration<Config_t, Rejection, 0b1> Rejection_t;  // Bit 8..10
+
+	struct FaultThreshold
+	{
+		typedef uint16_t RtdResistance_t;
+	};
+
+	static constexpr uint16_t
+	rtdfaultthresh(const max31865::FaultThreshold::RtdResistance_t res)
+	{
+		return static_cast<uint16_t>(res << 1);
+	};
+
+	enum class Register : uint8_t
+	{
+		ReadConfiguration = 0x00,
+		ReadRtdMsb = 0x01,
+		ReadRtdLsb = 0x02,
+		ReadHighFaultThresholdMsb = 0x03,
+		ReadHighFaultThresholdLsb = 0x04,
+		ReadLowFaultThresholdMsb = 0x05,
+		ReadLowFaultThresholdLsb = 0x06,
+		ReadFaultStatus = 0x07,
+
+		WriteConfiguration = 0x80,
+		WriteHighFaultThresholdMsb = 0x83,
+		WriteHighFaultThresholdLsb = 0x84,
+		WriteLowFaultThresholdMsb = 0x85,
+		WriteLowFaultThresholdLsb = 0x86,
+	};
+
+	struct modm_packed Data
+	{
+		/// @return measure resistance in ohm
+		constexpr float
+		getResistance() const
+		{
+			const uint16_t adccode = data >> 1;
+			return adccode / 32768.f * Rref;
+		}
+
+        /// @return fast temperature in degrees celsius, about 0.3 degrees error between 0 and 100 degrees celsius
+		constexpr float
+		getTemperatureFast() const
+		{
+			return (getResistance() - R0) / alpha / 100.f;
+		}
+
+		/// @return slow but accurate temperature in degrees celsius
+		constexpr double
+		getTemperaturePrecise() const
+		{
+			const double res = getResistance();
+			double T = getTemperatureFast();
+			const double c0 = T <= 0 ? c : 0;
+
+			// Do some fixed number of newton steps for root finding
+			// on Callendar Van Dusen equation:
+			// R = R0*(1+a*T+b*T*T+c*(T-100)*T*T*T)
+			// Newton seems to need double precision to achieve 1.e-10 residual?!
+			for (int i = 0; i < 10; i++)
+			{
+				const double R =
+					double{R0} * (1 + (a * T) + (b * T * T) + (c0 * (T - 100) * T * T * T)) - res;
+				const double Rdash =
+					double{R0} * (a + (2 * b * T) + c0 * (((4 * T) - 300) * T * T));
+				T -= R / Rdash;
+				if (std::abs(R) <= 1.e-10) { break; }
+			}
+
+			return T;
+		}
+
+		uint16_t data;
+	};
+};  // struct max31865
+
+using max31865pt100 = max31865<>;
```

Comment:
Wouldn't it make sense to put the default template arguments (R0, Rref, alpha, a, b, c) into a struct, pass it as a template argument to the `Max31865` class and forward it to the `max31865` struct?
That way the whole code is reusable and the `Max31865` class does not have to be copied to change the values.

Reply:
```c++

struct Max31865Config
{
	float R0;
	float Rref;
	float alpha;
	double a;
	double b;
	double c;
};

/// Standard values for IEC 751 (PT100) with 430 Ohm reference
constexpr Max31865Config max31865ConfigPt100
{
	.R0 = 100.f,
	.Rref = 430.f,
	.alpha = 3.85055e-3f,
	.a = double{3.90830e-3},
	.b = double{-5.77500e-7},
	.c= double{-4.18301e-12},
};

template<Max31865Config max31865Config>
struct max31865
{
	static constexpr float R0 = max31865Config.R0;
	static constexpr float Rref = max31865Config.Rref;
	// ...

	// [...]
};

template<typename SpiMaster, typename Cs, Max31865Config max31865Config = max31865ConfigPt100>
class Max31865 : public max31865<max31865Config>,
				 public modm::SpiDevice<SpiMaster>,
				 protected modm::NestedResumable<3>
{
	// [...]
};
```

Reply:
Should work like this.

Is `struct Pt{...}` is confusing in the modm namespace?
Should I call it something like `struct RtdPt{...}` maybe?